src/main/kotlin/io/github/zuccherosintattico/gradle/Typescript.kt

```diff
@@ -22,12 +23,16 @@ open class Typescript : Plugin<Project> {
             shouldInstall.set(nodeExtension.shouldInstall)
             zipUrl.set(nodeExtension.zipUrl)
             version.set(nodeExtension.version)
+            nodeBundleFile.set(project.nodeBundleFile())
+            projectDir.set(project.projectDir)
+            outputs.upToDateWhen { false } // Don't allow gradle to mark this task as UP-TO-DATE
```

Comment:
By adding an output file property it gave gradle the chance to mark the task as `UP-TO-DATE` so this blocks that from happening. If we were to rework the task further to correctly track all the output directories of the tools it installs we could probably simplify some of the logic in the task and just let gradle own the work avoidance logic.